app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,21 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      if params[:ems_type].blank?
+        all_options = Hash[
+          ExtManagementSystem.descendants.collect do |ems|
+            [ems.ems_type, ems.respond_to?(:provider_settings) ? ems.provider_settings : {}]
```

Comment:
I'm not sure I follow? I'd expect per-instance settings to be revealed via GETs to the collection member URI. We augment OPTIONS responses only as a limited way to reveal metadata - i.e. data about the resource itself in the abstract.